controllers/routing/routing_delete_resources.go

```diff
@@ -0,0 +1,75 @@
+package routing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/opendatahub-io/odh-platform/pkg/metadata"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// HandleResourceDeletion handles the removal of dependent resources when the target resource is being deleted.
+func (r *PlatformRoutingReconciler) HandleResourceDeletion(ctx context.Context, sourceRes *unstructured.Unstructured) (ctrl.Result, error) {
+	exportModes, found := extractExportModes(sourceRes)
+	if !found {
+		r.log.Info("No export modes found, skipping deletion logic", "sourceRes", sourceRes)
+
+		return ctrl.Result{}, nil
+	}
+
+	r.log.Info("Handling deletion of dependent resources", "sourceRes", sourceRes)
+
+	for _, exportMode := range exportModes {
+		if err := r.deleteResourcesByLabels(ctx, sourceRes, routingResourceGVKs(exportMode)); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to delete resources for export mode %s: %w", exportMode, err)
+		}
+	}
+
+	return removeFinalizerAndUpdate(ctx, r.Client, sourceRes)
+}
+
+func (r *PlatformRoutingReconciler) deleteResourcesByLabels(ctx context.Context, target *unstructured.Unstructured, gvkList []schema.GroupVersionKind) error {
```

Comment:
I don't think revealing implementation details (i.e. `ByLabels`) brings more clarity when reading the code for the first time. How about `deleteOwnedResources`?

Reply:
[538234e](https://github.com/maistra/odh-platform/pull/39/commits/538234edfd01f3afce213ad0fd4a80c19d5e342b) renamed